ste/xfer-remoteToLocal-file.go

```diff
@@ -417,6 +419,25 @@ func commonDownloaderCompletion(jptm IJobPartTransferMgr, info TransferInfo, ent
 			panic("reached branch where jptm is assumed to be live, but it isn't")
 		}
 
+		// Attempt to put MD5 data if necessary, compliant with the sync hash scheme
+		if jptm.ShouldPutMd5() {
+			fi, err := os.Stat(info.Destination)
+			if err != nil {
+				jptm.FailActiveDownload("saving MD5 data (stat to pull LMT)", err)
+				goto redoCompletion // let fail as expected
+			}
+
+			err = common.PutHashData(info.Destination, common.SyncHashData{
+				Mode: common.ESyncHashType.MD5(),
+				LMT:  fi.ModTime(),
+				Data: base64.StdEncoding.EncodeToString(info.SrcHTTPHeaders.ContentMD5),
+			})
+			if err != nil {
+				jptm.FailActiveDownload("saving MD5 data (writing alternate data stream)", err)
+				goto redoCompletion // let fail as expected
```

Comment:
Can we get into a loop here?

Reply:
Nope. At the point whe goto redoCompletion, the jptm is dead due to the failed download. If we wanted to, we could do a recursive call and drop the goto.